src/types/list.rs

```diff
@@ -83,17 +83,34 @@ impl PyList {
         }
     }
 
+    /// Takes the slice `self[low:high]` and returns it as a new list.
```

Comment:
Given that you're using the Python slice syntax here it may be worth calling out in docs that this doesn't take negative (from-the-end) indices.

(Maybe that's also worth commenting on at the top of this file on the `PyList` type?)

Reply:
Can do, although negative indices are out of the question due to the argument type already :)